sources/packages/backend/apps/api/src/route-controllers/student-appeal/student-appeal.controller.service.ts

```diff
@@ -15,13 +19,20 @@ export class StudentAppealControllerService {
    * - `assessDetails`, if true, will return access details.
    * @returns the student appeal and its requests.
    */
-  async getStudentAppealWithRequest<T>(
+  async getStudentAppealWithRequests<
```

Comment:
I am not sure if I follow this. The idea of the conditional return is to be used when the return type changes depending on the `T`. In this case, the `T` and `R` will be the same right?

Reply:
Mm, not sure, if I understood the question, I was following, https://dev.to/zirkelc/how-to-return-different-types-from-functions-in-typescript-2a2h (LAST EXAMPLE) https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABABwDYgM4B4AqiCmAHlPmACYaIZQBOMYA5ogD6JggC2ARvjQDSIASogC8iPERLlK1Oo0QB+KrXpMAXG048aAPgAUAQw04BXYwEoNwgN4BYAFCJEMYIj1QAnsnxxXB0SJiAOSyqkGIAGQRiJ7evohcAcGhjEHmiHaOTog0+FAgNEiGiADUCekGlIIA3A5OAL4Odc6u7l4+fkmIQezcvOFRMe3xiYHBvdppGc1OufmFbv5lXBVVtVmN9s1QABY0cADubPhHAKI0+zR6QQBCcLuIBjQMnKRQlByYUAn4iPG7vwwBg4v1i+DS602DggCGoOXwGBAqCgAEZRCh0Bg9CiBAAmczVRAAeiJ8MRyLRO0qQ28mj6NGhsO+uXJUFx6LQmGuAAl8KhUHBugIggB1OA0VBkCHE0kspFsxBUyhg5RyBhAA

Reply:
The example is potentially receiving 2 different types and conditionally returning a string or number based on the input type received.
The approach has a benefit when the generic type is not explicitly provided as below.
```ts
this.studentAppealControllerService.getStudentAppealWithRequests<DetailedStudentAppealRequestAPIOutDTO>(...);
```